### PR TITLE
4581 optional final Tanh

### LINKED ISSUE #4581: classification ViT have the final nn.Tanh()
### Discussed in https://github.com/Project-MONAI/MONAI/discussions/4580

<div type='discussions-op-text'>

<sup>Originally posted by **chris-clem** June 24, 2022</sup>
I was wondering why does the Vision Transformer (ViT) with `classification=True` have `nn.Tanh()` in the end of the `classification_head` (https://github.com/Project-MONAI/MONAI/blob/dev/monai/networks/nets/vit.py#L100)? The other classification models output the raw logits.

</div>

the hard-coded activation function should be made optional and documented.

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

Fixes #4581
making the final nonlinear optional

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/networks/nets/vit.py [hunk 1]
```diff
@@ -43,6 +43,7 @@ def __init__(
         num_classes: int = 2,
         dropout_rate: float = 0.0,
         spatial_dims: int = 3,
+        acti="Tanh",
```
Comment (finalelement) [inline monai/networks/nets/vit.py:46]: Not necessary to make this change, but perhaps using the term 'post_activation' might be more insightful to the user. Upto you. :)
